CODINGSTYLE.md

```diff
@@ -416,36 +416,57 @@ There is a check-script on the git server (also available for clients, see below
+* Revert: Reverting an earlier Feature / Add / Change / Fix / Remove.
+* Doc: Update to documentation.
+* Update: Translator commits.
+
+When a commit influence the moddable content, the above is also used, followed by a `[NewGRF]` or `[Script]` section.
+This means the commit is aimed (and worded) towards the NewGRF / Script authors, rather than players.
+
+Developer-facing changes:
+* Codechange: Changes to the code the player is not going to notice. Refactors, modernization, etc.
+* Cleanup: Similar to Codechange, but when it is more about removing old code, rather than an actual change.
+* Codefix: Fixing problems in earlier commits that the player is not actually going to notice. Wrong comments, missing files, CI changes.
+
+If you commit a `Fix` for an [issue](https://github.com/OpenTTD/OpenTTD/issues), add the corresponding issue number in the form of #NNNN.
+
+In the case of `Fix`es, if you know what commit hash the bug was introduced (eg regression), please mention that revision (the first 7 characters of the hash) as well just after the keyword.
+Finding the trouble-causing hash is highly encouraged as it makes backporting / branching / releases that much easier.
+
+Do not mention two keywords; if two apply, pick one that best represents the commit (for example, "Fix #123" is mostly always better than "Revert", even if both are true).
+
+The `<Details>` part starts with a capital and does not end with a dot.
+Try to be descriptive to what the player will notice, not to what is actually being changed in the code.
+See `changelog.txt` for inspiration.
 
 To further structure the changelog, you can add sections. Example are:
-* "Network" for network specific changes
-* "NewGRF" for NewGRF additions
-* "YAPP", "NPF", for changes in these features
-* "OSX", "Debian", "win32", for OS-specific packaging changes
+* "Network" for network specific changes.
+* "NewGRF" for NewGRF additions.
+* "Script" for AI / GS additions.
+* "YAPP", "NPF", for changes in these features.
+* "MacOS", "Linux", "Windows", for OS-specific changes.
+* "CI", "CMake", for changes to the (build) infrastructure.
 
-Further explanations, general bitching, etc. don't go into the first line. Use a new line for those.
+Further explanations, more details, etc. don't go into the first line. Use a new line for those.
 
 Complete examples:
-* Fix: [YAPF] Infinite loop in pathfinder.
-* Fix #5926: [YAPF] Infinite loop in pathfinder.
-* Fix 80dffae130: Warning about unsigned unary minus.
-* Fix #6673, 99bb3a95b4: Store the map variety setting in the samegame.
-* Revert d9065fbfbe, Fix #5922: ClientSizeChanged is only called via WndProcGdi which already has the mutex.
-* Fix #1264, Fix #2037, Fix #2038, Fix #2110: Rewrite the autoreplace kernel.
+* `Fix: [YAPF] Infinite loop in pathfinder`
+* `Fix #5926: [YAPF] Infinite loop in pathfinder`
+* `Codefix 80dffae: Warning about unsigned unary minus`
+* `Fix #6673, 99bb3a9: Store the map variety setting in the savegame`
```

Comment:
Should it be "commit, issue", or "issue, commit"? The example is "commit, issue", but text earlier says "revision [...] just after the keyword".

Reply:
Added some words for the example to be correct :) At least, I think that is the best, as the ticket is the more important piece of information (to the player).

Reply:
I'm fine with either way, as long as the words are consistent :)